src/controllers/tasks.controller.js

```diff
@@ -0,0 +1,183 @@
+/* eslint-disbled */
+import TaskModel from '../models/task.model.js';
+import faker from 'faker';
+
+export function findAllTasks() {
+  console.log('CONTROLLER: trying to get all the tasks ');
+  return TaskModel.find();
+}
+
+export function findTask(req, res) {
+  console.log(' CONTROLLER: get a task ');
+  const { id } = req.params;
+
+  TaskModel.findById(id)
+    .then((data) => {
+      console.log(data);
+      res.status(200).send(data);
+    })
+    .catch((err) => {
+      res.status(404).send({ message: err.message || 'task not found' });
+    });
+}
+
+export function search(req, res) {
+  console.log(' CONTROLLER: SEARCH  ');
+  const { phrase } = req.body;
+  console.log(phrase);
+  console.log('CONTROLLER: search');
+
+  TaskModel.find({ $text: { $search: phrase } })
+    .then((data) => {
+      console.log(data);
+      res.status(200).send(data);
+    })
+    .catch((err) => {
+      res.status(500).send({
+        message: err.message || 'Something went worng during the search..',
+      });
+    });
+}
+
+export function addTask(req, res) {
+  console.log('CONTROLLER: addtask ');
+  console.log(req.body.title);
+
+  // Validate request
+  if (!req.body) {
+    return res.status(400).send({
+      message: 'Please fill all required field',
+    });
+  }
+
+  const tareaNueva = new TaskModel({
+    title: req.body.title,
+    description: req.body.description,
+    phone: req.body.phone,
+  });
+
+  /*
+Tambien se puede pasar el objeto completo : 
+    const tareaNueva2 = new TaskModel(req.body);
+console.log("TAREA 2 " + tareaNueva2);
+
+*/
+  tareaNueva
+    .save()
+    .then((data) => {
+      console.log(data);
+      console.log('New Task inserted correctly..');
+      res.send(data);
+    })
+    .catch((err) => {
+      res.status(500).send({
+        message:
+          err.message ||
+          'Something went wrong while creating new task in the data base.',
+      });
+    });
+}
+
+export function updateTask(req, res) {
+  console.log(' CONTROLLER: triyin to update a task ');
+  const { id } = req.params;
+  const { description } = req.body;
+  const { phone } = req.body;
+  const { title } = req.body;
+
+  // TaskModel.update({ _id: id }, req.body);
+  console.log(description);
+  TaskModel.findById(id)
+    .then((data) => {
+      console.log(data);
+      data.description = description;
+      data.phone = phone;
+      data.title = title;
+      data
+        .save()
+        .then((data) => {
+          console.log('Task updated correctly.');
+          console.log(data);
+          res.status(200).send(data);
+        })
+        .catch((err) => res.status(500).send({ message: err.message }));
+    })
+    .catch((err) => {
+      res.status(500).send({ message: err.message });
+    });
```

Comment:
This is the beginning of the so-called promise hell (https://medium.com/@pyrolistical/how-to-get-out-of-promise-hell-8c20e0ab0513#:~:text=Unlike%20Callback%20Hell%2C%20Promise%20Hell%20is%20accidentally%20self%20inflicted.,snippets%20are%20written%20ECMAScript%202015.)

Since it's a promise, I'd use `async/await`.

I haven't used `mongoose` to be sure of this, but I think there must be another way to update a document 🤷